integration/write_read_timezone_test.go

```diff
@@ -0,0 +1,126 @@
+package integration
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	xtime "github.com/m3db/m3x/time"
+	"github.com/stretchr/testify/require"
+)
+
+type readWriteTZCase struct {
+	namespace  string
+	id         string
+	datapoints []readWriteTZDP
+}
+
+type readWriteTZDP struct {
+	value     float64
+	timestamp time.Time
+}
+
+// Make sure that everything works properly end-to-end even if the client issues
+// a write in a timezone other than that of the server.
+func TestWriteReadTimezone(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	// Ensure that the test is running with the local timezone set to US/Pacific
```

Comment:
could you put 1-2 lines explaining why you prefer doing this over changing the clock.Now